### PR TITLE
Split vanilla store + additions + kkiv 2035 mod support

### PR DESCRIPTION
**Pull request by wersal454**
## What type of PR is this.
1. [ ] Bug
2. [x] Change
3. [x] Enhancement

### What have you changed and why?
Information:

Splitted vanilla store into dlcs + added some stuff
Added support for kkiv 2035 mod    

### Please verify the following and ensure all checks are completed.

1. [ ] Have you loaded the mission in LAN host?
2. [ ] Have you loaded the mission on a dedicated server?

### Is further testing or are further changes required?
1. [ ] No
2. [x] Yes (Please provide further detail below.)

### How can the changes be tested?
Steps: 
enable dlcs(?)
load mod
go to blackmarket guy and check the store

### CODE REVIEW

**Comment by Silence-112** [comment]
I don't mind the kkiv 2035 mod support/additions part of this PR, but the Split vanilla store part is a no-go. It's pointless and will bloat the store far too much. 

Perhaps it might be better to close this PR and open a new one with just the additions?

**Comment by wersal454** [comment]
nuh, honestly I think problem is how categories are displayed in the store. I've looked at code for store ui, haven't figured out how to change listbox placement, but I will figure it out, because now it's kinda junky

**Comment by Silence-112** [comment]
At the very least do it in a separate PR

File: A3A/addons/hals/Addons/store/config/contact.hpp [hunk 1]
```diff
@@ -0,0 +1,382 @@
...
+			ITEM(H_HelmetSpecB_wdl, 900, MISC_STOCK);
+
+			ITEM(G_RegulatorMask_F, 200, MISC_STOCK);
+			ITEM(G_AirPurifyingRespirator_01_F, 200, MISC_STOCK);
+			ITEM(G_AirPurifyingRespirator_02_sand_F, 200, MISC_STOCK);
+			ITEM(G_AirPurifyingRespirator_02_olive_F, 200, MISC_STOCK);
+			ITEM(G_AirPurifyingRespirator_02_black_F, 200, MISC_STOCK);
+
+			ITEM(G_Blindfold_01_black_F, 200, MISC_STOCK);
+			ITEM(G_Blindfold_01_whitw_F, 200, MISC_STOCK);
```
Comment (Silence-112) [inline A3A/addons/hals/Addons/store/config/contact.hpp]:
```suggestion
			ITEM(G_Blindfold_01_white_F, 200, MISC_STOCK);

```

**Review by Silence-112 — CHANGES_REQUESTED** [review]
There's 2 missing categories (not sure which cfg they belong to)

May also be worth adding a "spacer" category, so that modsets have a single space between them. E.g:

```
Apex SMGs
Apex Snipers
\\ ======= // - empty spacer category
Art Of War Misc
\\ ======= // - empty spacer category
Etc
```

**Comment by wersal454** [comment]
> There's 2 missing categories (not sure which cfg they belong to)
> 
> May also be worth adding a "spacer" category, so that modsets have a single space between them. E.g:
> 
> ```
> Apex SMGs
> Apex Snipers
> \\ ======= // - empty spacer category
> Art Of War Misc
> \\ ======= // - empty spacer category
> Etc
> ```

Fixed it, I like idea with spacers though I have no idea how it could be done, if you know do a suggestion and I look into it

**Comment by Silence-112** [comment]
Spacers would have to be done via the function that initialises the trader categories, i'll look into that but there is a bigger issue currently

All of this stuff will still be added to the store even if you have mods loaded (previously if you had mods loaded, it would not add any vanilla stuff unless you explicitly enabled the param to do so)

We still want it to be that way, so will need to figure something out

File: A3A/addons/ultimate/config/trader/cfgTraderMods.hpp [hunk 2]
```diff
@@ -5,6 +5,51 @@ class traderMods
...
+    class marksmen : base
+    {
+        addons[] = {"A3_Weapons_F_Mark"};
+        prefix = "marksmen";
+    };
+    class tanks : base
+    {
+        addons[] = {"A3_Weapons_F_Tank"};
+        prefix = "tanks";
+    };
```
Comment (Silence-112) [inline A3A/addons/ultimate/config/trader/cfgTraderMods.hpp]:
```suggestion
```

**Review by Silence-112 — CHANGES_REQUESTED** [review]

File: A3A/addons/ultimate/config/trader/cfgTraderMods.hpp (hunk 2 above)
Comment (Silence-112) [reply A3A/addons/ultimate/config/trader/cfgTraderMods.hpp]: I pushed a commit, so that suggestion won't make much sense without reading it

**Comment by Silence-112** [comment]
Seems like too much work to get proper spacers to work, so this will do fine

**Review by Silence-112 — APPROVED** [review]

**Pull request merged**